Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1270,7 +1270,7 @@ void RenderLayer::recursiveUpdateLayerPositions(RenderElement::LayoutIdentifier
         dest = source; \
     else \
         LAYER_POSITIONS_ASSERT(dest == source);
-    UPDATE_OR_VERIFY_STATE_BIT(m_repaintStatus, RepaintStatus::NeedsNormalRepaint);
+    UPDATE_OR_VERIFY_STATE_BIT(this->m_repaintStatus, RepaintStatus::NeedsNormalRepaint);
```

Comment:
So is this happening with GCC 14? If so, I will try to reproduce and report a bug.

Reply:
Yes.

Reply:
Typically when we have code to work around a compiler bug, we add a comment so that someone well meaning doesn’t just change the code back. Related: do we have this version of GCC in EWS? If not, how are we going to prevent someone from accidentally changing it back?

Reply:
The affected code seems to run only inside `ASSERT()` calls. We don't currently have a EWS bot with asserts enabled (no Debug assert EWS bots, only in post-commit), but work is underway to enable asserts in the WPE EWS queues, to  prevent problems like this (and assert crashes in Linux ports!) from sneaking in.

Reply:
Tested with GTK port build of 285131@main with the change from this pull request reverted to reintroduce the bug, using gcc-14.2.1-3.fc41. Unfortunately I'm not able to reproduce the problem.

[Instructions here](https://gcc.gnu.org/bugs/) if somebody else wants to try reporting a bug. You just need to collect the preprocessed source file using -save-temps so the bug report only depends on a single file rather than the entire WebKit build system.